modules/RTC/CodecSelection.js

```diff
@@ -32,24 +32,29 @@ export class CodecSelection {
     constructor(conference, options) {
         this.conference = conference;
         this.options = options;
-
-        // VP8 cannot be disabled and it will be the default codec when no preference is set.
-        this.disabledCodec = options.disabledCodec === CodecMimeType.VP8
-            ? undefined
-            : this._getCodecMimeType(options.disabledCodec);
-
-        // Check if the codec values passed are valid.
-        const jvbCodec = this._getCodecMimeType(options.jvbCodec);
-        const p2pCodec = this._getCodecMimeType(options.p2pCodec);
-
-        this.jvbPreferredCodec = jvbCodec && this._isCodecSupported(jvbCodec) ? jvbCodec : CodecMimeType.VP8;
-        this.p2pPreferredCodec = p2pCodec && this._isCodecSupported(p2pCodec) ? p2pCodec : CodecMimeType.VP8;
-        logger.debug(`Codec preferences for the conference are JVB: ${this.jvbPreferredCodec},
-            P2P: ${this.p2pPreferredCodec}`);
-
-        if (this.jvbPreferredCodec === CodecMimeType.VP9 && !browser.supportsVP9()) {
-            this.jvbPreferredCodec = CodecMimeType.VP8;
-        }
+        this.enforcePreferredCodec = options.enforcePreferredCodec;
+
+        // VP8 cannot be disabled since it the default codec.
+        this.p2pDisabledCodec = options.p2pDisabledCodec !== CodecMimeType.VP8
+            && this._isCodecSupported(options.p2pDisabledCodec)
```

Comment:
Why do we need to support the codec in order to disable it?

Reply:
I am checking it here to avoid unnecessary renegotiations. `SDPUtil.stripCodec` and` SDPUtil.preferCodec` methods don't return a value indicating if the order was actually changed and calling setVideoCodecs triggers a renegotiation automatically.